tests/foreman/ui/test_computeresource_azurerm.py

```diff
@@ -0,0 +1,241 @@
+            assert azurecloud_vm
+            assert azurecloud_vm.is_running
+            assert azurecloud_vm.name == hostname.lower()
+            assert azurecloud_vm.ip == host_info['properties']['properties_table']['IP Address']
+            assert azurecloud_vm.type == AZURERM_VM_SIZE_DEFAULT
+
+            # Host Delete
+            session.host.delete(fqdn)
+            assert not session.host.search(fqdn)
+
+            # AzureRm Cloud assertion
+            assert not azurecloud_vm.exists
+
+        except Exception as error:
+            azure_vm = entities.Host().search(query={'search': 'name={}'.format(fqdn)})
+            if azure_vm:
+                azure_vm[0].delete()
+            raise error
+
+        finally:
+            skip_yum_update_during_provisioning(template='Kickstart default finish', reverse=True)
+
+
+@tier3
+@upgrade
+def test_positive_azurerm_host_provision_ud(
+    session,
+    azurermclient,
+    module_azurerm_cloudimg,
+    module_azurerm_cr,
+    module_domain,
+    module_os,
+    module_org,
+    module_loc,
+    module_azure_hg,
+):
+
+    """Provision a Host with hostgroup and Compute-profile using
+    cloud-init image on AzureRm compute resource
+
+    :id: 2dc6c494-0e80-4845-af8f-43d37f69a093
+
+    :expectedresults: Host is provisioned successfully.
+
+    :CaseImportance: Critical
+
+    :CaseLevel: System
+    """
+
+    hostname = gen_string('alpha')
+    fqdn = '{}.{}'.format(hostname, module_domain.name).lower()
+
+    with session:
+
+        # Provision Host
+        try:
+            skip_yum_update_during_provisioning(template='Kickstart default user data')
+            session.host.create(
+                {
+                    'host.name': hostname,
+                    'host.hostgroup': module_azure_hg.name,
+                    'provider_content.operating_system.root_password': gen_string('alpha'),
+                    'provider_content.operating_system.image': module_azurerm_cloudimg.name,
+                }
+            )
+
+            host_info = session.host.get_details(fqdn)
+            assert host_info['properties']['properties_table']['Build'] == 'Pending installation'
+            assert (
+                host_info['properties']['properties_table']['Host group'] == module_azure_hg.name
+            )
+
+            # AzureRm Cloud assertion
+            azurecloud_vm = azurermclient.get_vm(name=hostname.lower())
+            assert azurecloud_vm
+            assert azurecloud_vm.is_running
+            assert azurecloud_vm.name == hostname.lower()
+            assert azurecloud_vm.ip == host_info['properties']['properties_table']['IP Address']
+            assert azurecloud_vm.type == AZURERM_VM_SIZE_DEFAULT
+
+        except Exception as error:
+            azure_vm = entities.Host().search(query={'search': 'name={}'.format(fqdn)})
+            if azure_vm:
+                azure_vm[0].delete()
+            raise error
+
```

Comment:
I'm trying to understand here that if you wanted to raise anyway, you could have only had try and finally, by that way you no need to raise the error and checking again and again about the  `azure_vm = entities.Host().search(query={'search': 'name={}'.format(fqdn)})    `

I might be wrong, I'm not able to understand that except block need.

Reply:
Probably a copy/paste error but not sure why i kept it under same ... I have removed that now as not require :)